spring-resttemplate-2/src/main/java/com/baeldung/resttemplate/json/service/UserServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package com.baeldung.resttemplate.json.service;
+
+import com.baeldung.resttemplate.json.model.Address;
+import com.baeldung.resttemplate.json.model.User;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Override
+    public List<User> getUsers() {
+        return List.of(
+                new User(1, "user1", new ArrayList<Address>(
```

Comment:
It might be cleaner to use `Arrays.asList` here, statically importing it. You don't need to assign that to a new `ArrayList`.